sources/packages/web/src/components/common/StudentApplication.vue

```diff
@@ -129,50 +132,60 @@ export default defineComponent({
-          formInstance,
-          LOCATIONS_DROPDOWN_KEY,
+      if (formInstance && props.initialData) {
+        formData.value = props.initialData;
+        offeringIntensity = formData.value.applicationOfferingIntensityValue;
+        const applicationIntensityComponent = formInstance.getComponent(
+          OFFERING_INTENSITY_KEY,
         );
-        const selectedLocationId = getSelectedId(formInstance);
-
-        if (selectedLocationId) {
-          // when isReadOnly.value is true, then consider
-          // both active and inactive program year.
-          await formioDataLoader.loadProgramsForLocation(
-            formInstance,
-            +selectedLocationId,
-            PROGRAMS_DROPDOWN_KEY,
-            props.programYearId,
-            props.isReadOnly,
-          );
+        // Program year forms older than 2025-26 required offering intensity to be injected to the form.
+        // Check if the form has the offering intensity component and if so, inject the values.
+        if (applicationIntensityComponent) {
+          formData.value.howWillYouBeAttendingTheProgram =
+            props.initialData.applicationOfferingIntensityValue;
         }
-        const selectedProgramId = formioUtils.getComponentValueByKey(
-          formInstance,
-          PROGRAMS_DROPDOWN_KEY,
-        );
-        const selectedIntensity: OfferingIntensity =
-          formioUtils.getComponentValueByKey(
+        // When the form is editable, load locations and programs.
+        if (!props.isReadOnly) {
+          await formioDataLoader.loadLocations(
             formInstance,
-            OFFERING_INTENSITY_KEY,
+            LOCATIONS_DROPDOWN_KEY,
           );
-        if (selectedProgramId && selectedIntensity) {
-          await formioComponentLoader.loadProgramDesc(
-            formInstance,
-            selectedProgramId,
-            SELECTED_PROGRAM_DESC_KEY,
-          );
-          // when isReadOnly.value is true, then consider
-          // both active and inactive program year.
-          await formioDataLoader.loadOfferingsForLocation(
+          const selectedLocationId = getSelectedId(formInstance);
+
+          if (selectedLocationId) {
+            // when isReadOnly.value is true, then consider
+            // both active and inactive program year.
+            await formioDataLoader.loadProgramsForLocation(
+              formInstance,
+              +selectedLocationId,
+              PROGRAMS_DROPDOWN_KEY,
+              props.programYearId,
+              props.isReadOnly,
+            );
+          }
+          const selectedProgramId = formioUtils.getComponentValueByKey(
             formInstance,
-            selectedProgramId,
-            selectedLocationId,
-            OFFERINGS_DROPDOWN_KEY,
-            props.programYearId,
-            selectedIntensity,
-            props.isReadOnly,
+            PROGRAMS_DROPDOWN_KEY,
           );
+
+          if (selectedProgramId && offeringIntensity) {
```

Comment:
There is no longer a scenario where `offeringIntensity` won't be present, the `offeringIntensity` can be removed from the `if`.